### PR TITLE
fix #4171 feat(nimbus): use nimbus_id for fenix experiments

### LINKED ISSUE #4171: Use `nimbus_id` instead of `normandy_id` for fenix experiments
The DTO serializer sets the `randomization_unit` to `normandy_id` for all experiments right now, but it needs to be `nimbus_id` for fenix experiments.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-696)
┆Issue Number: EXP-696

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* Fenix clients using the Rust SDK require the randomization unit to be set to `nimbus_id` not `normandy_id`

This commit

* Adds per application randomization units
* Uses `normandy_id` for desktop
* Uses `nimbus_id` for fenix

### CODE REVIEW

File: app/experimenter/nimbus-ui/schema.graphql [hunk 1]
```diff
@@ -177,12 +177,17 @@ type NimbusFeatureConfigType {
   nimbusexperimentSet: [NimbusExperimentType!]!
 }
 
+enum NimbusIsolationGroupRandomizationUnit {
```
Comment (k88hudson) [inline app/experimenter/nimbus-ui/schema.graphql:180]: Why isolationgroup?

**Comment by k88hudson** [comment]
Looks good although the names of things seem a big weird to me

**Review by k88hudson — APPROVED** [review]
R+

File: app/experimenter/nimbus-ui/schema.graphql (hunk 1 above)
Comment (jaredlockhart) [reply app/experimenter/nimbus-ui/schema.graphql:180]: Why not isolation group? 🤔 

Also becuase randomization unit is a property of isolation group:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/models/nimbus.py#L212

**Pull request merged**